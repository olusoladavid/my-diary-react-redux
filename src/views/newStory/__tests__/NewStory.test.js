import thunk from "redux-thunk";
import configureStore from "redux-mock-store";

import ConnectedNewStory, { NewStory } from "../NewStory";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const storeState = {
  newStory: {
    loading: false,
    created: true
  }
};

const props1 = {
  hideToast: jest.fn(),
  createStory: jest.fn(),
  showToast: jest.fn(),
  loading: false,
  newStory: jest.fn(),
  created: false
};

const props2 = {
  ...props1,
  created: true
};

const props3 = {
  ...props1,
  loading: true
};

it("renders without crashing", () => {
  const wrapper = shallow(<NewStory {...props1} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing when story is created", () => {
  const wrapper = shallow(<NewStory {...props2} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing when story is being created", () => {
  const wrapper = shallow(<NewStory {...props3} />);
  expect(wrapper).toMatchSnapshot();
});

it("clears errors when input fields are focused", () => {
  const wrapper = shallow(<NewStory {...props1} />);
  wrapper
    .find(".form__input")
    .at(0)
    .simulate("focus");
  expect(props1.hideToast).toHaveBeenCalled();
});

it("should save user input to state", () => {
  const wrapper = shallow(<NewStory {...props1} />);
  wrapper
    .find(".js-new-story-title")
    .at(0)
    .simulate("change", { target: { value: "title", name: "title" } });
  expect(wrapper.state("title")).toEqual("title");
});

it("should not submit form when title is lesser than 5 chars", () => {
  const wrapper = shallow(<NewStory {...props1} />);
  wrapper.setState({ title: "t", content: "c" });
  wrapper
    .find(".js-new-story")
    .at(0)
    .simulate("submit", { preventDefault: jest.fn() });
  expect(props1.createStory).not.toHaveBeenCalled();
});

it("should synchronise with app state in the store", () => {
  const store = mockStore(storeState);
  const wrapper = shallow(<ConnectedNewStory {...props1} store={store} />);
  expect(wrapper.prop("created")).toBe(true);
});
