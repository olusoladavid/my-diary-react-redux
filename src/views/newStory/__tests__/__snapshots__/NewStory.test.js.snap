// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="form-group js-new-story"
    >
      <h2
        className="form-group__heading"
      >
        05 DECEMBER 2018
      </h2>
      <form
        action="#"
        className="form-group__form"
        onSubmit={[Function]}
      >
        <input
          className="form__input js-new-story-title"
          name="title"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Title"
          required={true}
          type="text"
          value=""
        />
        <textarea
          className="form__input js-new-story-content"
          cols={10}
          name="content"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Content"
          required={true}
          rows={12}
          value=""
        />
        <button
          className="form__btn--submit js-new-story-btn"
          type="submit"
        >
          Save
        </button>
        <p
          className="form__error js-error-field"
        />
      </form>
    </div>
  </div>
</Fragment>
`;

exports[`renders without crashing when story is being created 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="form-group js-new-story"
    >
      <h2
        className="form-group__heading"
      >
        05 DECEMBER 2018
      </h2>
      <form
        action="#"
        className="form-group__form"
        onSubmit={[Function]}
      >
        <input
          className="form__input js-new-story-title"
          name="title"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Title"
          required={true}
          type="text"
          value=""
        />
        <textarea
          className="form__input js-new-story-content"
          cols={10}
          name="content"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Content"
          required={true}
          rows={12}
          value=""
        />
        <button
          className="form__btn--submit js-new-story-btn"
          type="submit"
        >
          <i
            className="fas fa-circle-notch fa-spin"
          />
        </button>
        <p
          className="form__error js-error-field"
        />
      </form>
    </div>
  </div>
</Fragment>
`;

exports[`renders without crashing when story is created 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="form-group js-new-story"
    >
      <h2
        className="form-group__heading"
      >
        05 DECEMBER 2018
      </h2>
      <form
        action="#"
        className="form-group__form"
        onSubmit={[Function]}
      >
        <input
          className="form__input js-new-story-title"
          name="title"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Title"
          required={true}
          type="text"
          value=""
        />
        <textarea
          className="form__input js-new-story-content"
          cols={10}
          name="content"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Content"
          required={true}
          rows={12}
          value=""
        />
        <button
          className="form__btn--submit js-new-story-btn"
          type="submit"
        >
          Save
        </button>
        <p
          className="form__error js-error-field"
        />
      </form>
    </div>
  </div>
</Fragment>
`;
